internal/merger/merger.go

```diff
@@ -34,14 +34,14 @@ func MergeObservation(
 					OrderedFacets: []*pbv2.FacetObservation{},
 				}
 			}
-			eData.OrderedFacets = append(
-				eData.OrderedFacets,
-				o1.ByVariable[v].ByEntity[e].OrderedFacets...,
+			o1.ByVariable[v].ByEntity[e].OrderedFacets = append(
+				o1.ByVariable[v].ByEntity[e].OrderedFacets,
+				eData.OrderedFacets...,
 			)
 		}
 	}
 	for facetID, facet := range o2.Facets {
 		o1.Facets[facetID] = facet
 	}
-	return o1, nil
+	return o1
```

Comment:
since this always return nil, no need to return error?

Reply:
Right, that's what this line's change is for.